### PR TITLE
Read non english character from runsettings

### PR DESCRIPTION
**Pull request by Faizan2304**
Issue: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/510768 
Currently TestPlatform throw exception ("Settings file provided does not conform to required format. Invalid character in the given encoding") if runsettings has Japanese character.

Fix: Read contents of file using UTF-8 encoding. Customer has to make sure that he has his runsettings in UTF-8 format.

### CODE REVIEW

File: src/vstest.console/Processors/RunSettingsArgumentProcessor.cs [hunk 1]
```diff
@@ -165,7 +166,11 @@ private void ExtractFrameworkAndPlatform()
             Justification = "XmlReaderSettings.XmlResolver is not available in core. Suppress until fxcop issue is fixed.")]
         protected virtual XmlReader GetReaderForFile(string runSettingsFile)
         {
-            return XmlReader.Create(runSettingsFile, XmlRunSettingsUtilities.ReaderSettings);
+            using (var fileStream = File.OpenRead(runSettingsFile))
+            {
+                var stream = new StreamReader(fileStream, Encoding.UTF8);
```
Comment (smadala) [inline src/vstest.console/Processors/RunSettingsArgumentProcessor.cs]: Use using for `stream ` too.

File: test/vstest.console.UnitTests/Processors/RunSettingsArgumentProcessorTests.cs [hunk 2]
```diff
@@ -291,6 +292,51 @@ public void InitializeShouldNotUpdateCommandLineOptionsArchitectureAndFxIfNotPro
             Assert.IsFalse(CommandLineOptions.Instance.FrameworkVersionSpecified);
         }
 
+        [TestMethod]
+        public void InitializeShouldNotThrowExceptionIfRunSettingsHasJapaneseCharacter()
+        {
+            var runsettingsFile = Path.Combine(Path.GetTempPath(), "InitializeShouldNotThrowExceptionIfRunSettingsHasJapaneseCharacter.runsettings");
+            var settingsXml = @"<RunSettings><RunConfiguration><ResultsDirectory>C:\新しいフォルダー</ResultsDirectory></RunConfiguration></RunSettings>";
+
+            if (File.Exists(runsettingsFile))
```
Comment (smadala) [inline test/vstest.console.UnitTests/Processors/RunSettingsArgumentProcessorTests.cs]: Check is not required.

**Review by smadala — COMMENTED** [review]
Verify commandline scenario -- RunConfiguration.ResultsDirectory=C:\新しいフォルダー

File: test/vstest.console.UnitTests/Processors/RunSettingsArgumentProcessorTests.cs [hunk 3]
```diff
@@ -291,6 +292,51 @@ public void InitializeShouldNotUpdateCommandLineOptionsArchitectureAndFxIfNotPro
...
+                CommandLineOptions.Instance,
+                this.settingsProvider,
+                null);
+
+            executor.Initialize(runsettingsFile);
+            File.Delete(runsettingsFile);
+        }
+
+        [TestMethod]
+        public void InitializeShouldPreserveActualJapaneseString()
```
Comment (smadala) [inline test/vstest.console.UnitTests/Processors/RunSettingsArgumentProcessorTests.cs]: One test is sufficient.

File: test/vstest.console.UnitTests/Processors/RunSettingsArgumentProcessorTests.cs (hunk 2 above)
Comment (Faizan2304) [reply test/vstest.console.UnitTests/Processors/RunSettingsArgumentProcessorTests.cs]: Resolved

**Review by smadala — APPROVED** [review]

File: test/vstest.console.UnitTests/Processors/RunSettingsArgumentProcessorTests.cs (hunk 3 above)
Comment (Faizan2304) [reply test/vstest.console.UnitTests/Processors/RunSettingsArgumentProcessorTests.cs]: Deleted other one

**Review by singhsarab — APPROVED** [review]

**Comment by Faizan2304** [comment]
@dotnet-bot  test Windows_NT / Debug Build 
@dotnet-bot  test Windows_NT / Release Build please

**Pull request merged**